tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -1319,6 +1319,81 @@ def test_spilu_0(self, dtype, xp, sp):
         return ainv.solve(b)
 
 
+@testing.parameterize(*testing.product({
+    'damp': [0.0, 1.0, 2.0],
+    'format': ['coo', 'csr', 'csc'],
+    'm': [30, 40, 50],
+    'n': [20, 30],
+    'x0': [None, 'ones'],
+    'use_linear_operator': [False, True],
+}))
+@testing.with_requires('scipy')
+class TestLsmr(unittest.TestCase):
+
+    density = 0.01
+
+    def _make_matrix(self, xp):
+        shape = (self.m, self.n)
+        a = testing.shaped_random(shape, xp, scale=1)
+        mask = testing.shaped_random(shape, xp, scale=1)
+        a[mask > self.density] = 0
+        return a
+
+    def _make_normalized_vector(self, xp):
+        b = testing.shaped_random((self.m,), xp, scale=1)
+        return b / xp.linalg.norm(b)
+
+    def _test_lsmr(self, xp, sp, a):
+        b = self._make_normalized_vector(xp)
+        x0 = None
+        if self.x0 == 'ones':
+            x0 = xp.ones((self.n,))
+        return sp.linalg.lsmr(a, b, x0=x0, damp=self.damp)
+
+    @testing.numpy_cupy_allclose(rtol=1e-1, atol=1e-1, sp_name='sp')
+    def test_sparse(self, xp, sp):
+        if runtime.is_hip and self.format == 'csc':
+            pytest.xfail('may be buggy')  # trans=True
+        if (self.damp == 0 and self.x0 == 'ones' and self.n != 20):
+            raise unittest.SkipTest
+        a = self._make_matrix(xp)
+        a = sp.coo_matrix(a).asformat(self.format)
+        if self.use_linear_operator:
+            a = sp.linalg.aslinearoperator(a)
+        return self._test_lsmr(xp, sp, a)
```

Comment:
You compare each element of the tuples that SciPy's and CuPy's`lsmr` return, causing the CI to fail because of a larger error in some values in the tuples. I think comparing the least-square solution returned is enough.

```suggestion
        return self._test_lsmr(xp, sp, a)[0]
```